pkg/auth/auth_assisted_service_mock_test.go

```diff
@@ -81,73 +83,59 @@ func (f fakeInventory) DownloadMinimalInitrd(ctx context.Context, params install
 }
 
 func (f fakeInventory) DownloadClusterFiles(ctx context.Context, params installer.DownloadClusterFilesParams) middleware.Responder {
-	file, err := ioutil.TempFile("/tmp", "test.file")
+	file, err := os.CreateTemp("", "test")
 	if err != nil {
 		return installer.NewDownloadClusterFilesInternalServerError().WithPayload(
 			common.GenerateError(http.StatusInternalServerError, err))
 	}
+
+	defer os.Remove(fmt.Sprintf("{%s}/{%s}", "/tmp", file.Name()))
```

Comment:
This will give you `/tmp/tmp/test.file<random>`

See https://pkg.go.dev/os@go1.17.8#CreateTemp for details.
`Name()` is the fully qualified path, not local.

Reply:
changed to 
```go
defer os.Remove(file.Name())
```